src/mainboard/amd/romecrb/src/acpi.rs

```diff
@@ -11,44 +11,51 @@ use util::round_up_4k;
     let facs_offset = fadt_offset + size_of::<AcpiTableFadt>();
     let dsdt_offset = facs_offset + size_of::<AcpiTableFacs>();
+
+    // let madt_offset = dsdt_offset + 69887;
     let madt_offset = dsdt_offset + DSDT_DSDTTBL_HEADER.len();
     let madt_local_apic_offset = madt_offset + size_of::<AcpiTableMadt>();
-    let io_apic_offset = madt_local_apic_offset + cores as usize * size_of::<AcpiMadtLocalApic>();
-    let local_x2apic_offset = io_apic_offset + size_of::<AcpiMadtIoApic>();
-    let local_lapicnmi_offset = local_x2apic_offset + cores as usize * size_of::<AcpiMadtLocalX2apic>();
-    let local_isor_offset = local_lapicnmi_offset + size_of::<AcpiMadtLocalX2ApicNMI>();
-    let mcfg_offset = local_isor_offset + 2 * size_of::<AcpiMadtInterruptOverride>();
-    let total_size = mcfg_offset + size_of::<AcpiTableMcfg>() - start;
+    let madt_local_x2apic_offset = madt_local_apic_offset + cores as usize * size_of::<AcpiMadtLocalApic>();
+    let madt_local_x2apic_nmi_offset = madt_local_x2apic_offset + cores as usize * size_of::<AcpiMadtLocalX2Apic>();
+    let madt_io_apic_offset = madt_local_x2apic_nmi_offset + size_of::<AcpiMadtLocalX2ApicNMI>();
+    let madt_local_isor_offset = madt_io_apic_offset + size_of::<AcpiMadtIoApic>();
+
+    let mcfg_offset = madt_local_isor_offset + 2 * size_of::<AcpiMadtInterruptOverride>();
+    let hpet_offset = mcfg_offset + size_of::<AcpiTableMcfg>();
+    let total_size = hpet_offset + size_of::<AcpiTableHpet>() - start;
 
     // setup rsdp - Root System Description Pointer
     let rsdp = AcpiTableRsdp { signature: SIG_RSDP, revision: 2, length: 36, xsdt_physical_address: xsdt_offset as u64, ..Default::default() };
+
     write!(w, "Write rsdp  at {:x?} \r\n", rsdp_offset).unwrap();
     write(w, rsdp, rsdp_offset, 0);
     write(w, gencsum(rsdp_offset, rsdp_offset + ACPI_RSDP_CHECKSUM_LENGTH), rsdp_offset, ACPI_RSDP_CHECKSUM_OFFSET); // XXX
     debug_assert_eq!(acpi_tb_checksum(rsdp_offset, rsdp_offset + ACPI_RSDP_CHECKSUM_LENGTH), 0);
+
     write(w, gencsum(rsdp_offset, rsdp_offset + ACPI_RSDP_XCHECKSUM_LENGTH), rsdp_offset, ACPI_RSDP_XCHECKSUM_OFFSET); // XXX
     debug_assert_eq!(acpi_tb_checksum(rsdp_offset, rsdp_offset + ACPI_RSDP_XCHECKSUM_LENGTH), 0);
 
     // xsdt - Extended System Description Table
     let xsdt_total_length = size_of::<AcpiTableHeader>() + size_of::<u64>() * NUM_XSDT_ENTRIES;
-    let xsdt = AcpiTableHeader { signature: SIG_XSDT, length: xsdt_total_length as u32, ..AcpiTableHeader::new() };
+    let xsdt = AcpiTableHeader { signature: SIG_XSDT, length: xsdt_total_length as u32, revision: 1, ..AcpiTableHeader::new() };
     write(w, xsdt, xsdt_offset, 0);
     // xsdt entries
     let mut xsdt_entries: [u64; NUM_XSDT_ENTRIES] = [0; NUM_XSDT_ENTRIES];
     xsdt_entries[0] = fadt_offset as u64;
     xsdt_entries[1] = madt_offset as u64;
-    // xsdt_entries[2] = mcfg_offset as u64;
+    xsdt_entries[2] = mcfg_offset as u64;
+    xsdt_entries[3] = hpet_offset as u64;
```

Comment:
Could these 4 lines be in an array literal, so xsdt_entries doesn't need to be mut?